python_ta/cfg/visitor.py

```diff
@@ -435,3 +442,28 @@ def _get_raise_exc(node: nodes.Raise) -> str:
         return f"{nodes.Raise.__name__} {next(exceptions).name}"
     except StopIteration:
         return nodes.Raise.__name__
+
+
+def _get_preconditions_node(func: nodes.FunctionDef) -> Optional[nodes.NodeNG]:
+    """A helper method that takes in a function definition node, retrieves its preconditions, and then parses them
+    into a AST node representing all the preconditions combined in an and statement. Returns None if there are no
+    preconditions."""
+    assertions = parse_assertions(func)
+    valid_assertions = [assertion for assertion in assertions if _is_python_precondition(assertion)]
+    # for now, return None if at least one non-valid assertion
+    # remove the length comparison and improve valid precondition checker to change this behaviour
+    if not valid_assertions or len(valid_assertions) != len(assertions):
+        return None
+    precondition_string = " and ".join(valid_assertions)
```

Comment:
One other thing, you should really wrap each assertion in parentheses, to avoid any issues regarding `and` and [operator precedence](https://docs.python.org/3/reference/expressions.html#operator-precedence). It's the same issue as C macros that you learned about in CSC209.